nautilus_trader/adapters/dydx/execution.py

```diff
@@ -1090,7 +1098,10 @@ async def _submit_order_single(self, order: Order) -> None:
         if order.order_type == OrderType.LIMIT:
             price = order.price.as_double()
         elif order.order_type == OrderType.MARKET:
-            price = 0
+            if order.side == OrderSide.BUY:
```

Comment:
I think we need to make this more generic. This the explanation from support of dydx:

There are a few possible reasons why the SDK allows setting a price for a market order:

1. Price slippage protection: By setting a price for a market order, you can limit the potential price slippage. For example, if you set a price of $100 for a market buy order, the order will only be executed if the market price is at or below $100. If the market price is above $100, the order will not be executed.

Reply:
yes actually I just added this to test if it working or not.
for a fix we can have a oracle price with +- 0.1% (configurable) slippage.